core/packet.h

```diff
@@ -232,108 +242,86 @@ class alignas(64) Packet {
   static void Free(PacketBatch *batch) { Free(batch->pkts(), batch->cnt()); }
 
  private:
-  typedef void *MARKER[0];     // generic marker for a point in a structure
-  typedef uint8_t MARKER8[0];  // generic marker with 1B alignment
-
   union {
-    // This is all lifted from rte_mbuf.h
     struct {
-      MARKER cacheline0_;
+      // offset 0: Virtual address of segment buffer.
+      void *buf_addr_;
+      // offset 8: Physical address of segment buffer.
+      alignas(8) phys_addr_t buf_physaddr_;
 
-      void *buf_addr_;            // Virtual address of segment buffer.
-      phys_addr_t buf_physaddr_;  // Physical address of segment buffer.
+      union {
+        __m128i rearm_data_;
 
-      uint16_t buf_len_;  // Length of segment buffer.
+        struct {
+          // offset 16:
+          alignas(8) uint16_t data_off_;
 
-      // next 6 bytes are initialised on RX descriptor rearm
-      MARKER8 rearm_data_;
-      uint16_t data_off_;
+          // offset 18:
+          uint16_t refcnt_;
 
-      /**
-       * 16-bit Reference counter.
-       * It should only be accessed using the following functions:
-       * rte_mbuf_refcnt_update(), rte_mbuf_refcnt_read(), and
-       * rte_mbuf_refcnt_set(). The functionality of these functions (atomic,
-       * or non-atomic) is controlled by the CONFIG_RTE_MBUF_REFCNT_ATOMIC
-       * config option.
-       */
-      union {
-        rte_atomic16_t refcnt_atomic_;  // Atomically accessed refcnt
-        uint16_t refcnt_;               // Non-atomically accessed refcnt
-      };
-      uint8_t nb_segs_;  // Number of segments.
-      uint8_t port_;     // Input port.
+          // offset 20:
+          uint16_t nb_segs_;  // Number of segments
 
-      uint64_t offload_flags_;  // Offload features.
-
-      // remaining bytes are set on RX when pulling packet from descriptor
-      MARKER rx_descriptor_fields1_;
+          // offset 22:
+          uint16_t _dummy0_;  // rte_mbuf.port
+          // offset 24:
+          uint64_t _dummy1_;  // rte_mbuf.ol_flags
```

Comment:
Yes, it was meant to be "do not access". Given that PMD drivers support a variety set of different offloading features (and most vectorized PMDs do not support any), I thought it would be safe to keep it unset for Packet<->rte_mbuf compatibility.